src/v/s3/client.h

```diff
@@ -61,6 +61,48 @@ class object_tag_formatter {
     ss::sstring _tags;
 };
 
+enum class error_outcome {
```

Comment:
Note for reviewer: a subsequent commit moves the `error_outcome` definitions into a `types` header file.